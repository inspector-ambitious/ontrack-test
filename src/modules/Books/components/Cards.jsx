import React from "react";
import PropTypes from "prop-types";
import { Row, Col, Card, ListGroup, ListGroupItem } from "react-bootstrap";

const Cards = ({ books }) => {
  return (
    <Row className="mt-4">
      {books.map(
        ({
          book_title,
          book_author,
          book_pages,
          book_publication_city,
          book_publication_country,
          book_publication_year,
          id,
        }) => (
          <Col
            key={id}
            xs={12}
            md={6}
            lg={3}
            className="d-flex align-items-stretch mb-4"
          >
            <Card className="flex-fill">
              <Card.Body>
                <Card.Title>{book_title}</Card.Title>
              </Card.Body>
              <ListGroup className="list-group-flush">
                <ListGroupItem>
                  Author: {book_author.map((bookAuthor, index) => bookAuthor)}
                </ListGroupItem>
                <ListGroupItem>Pages: {book_pages}</ListGroupItem>
                <ListGroupItem>
                  Publication City: {book_publication_city}
                </ListGroupItem>
                <ListGroupItem>
                  Publication Country: {book_publication_country}
                </ListGroupItem>
                <ListGroupItem>
                  Publication Year: {book_publication_year}
                </ListGroupItem>
              </ListGroup>
            </Card>
          </Col>
        )
      )}
    </Row>
  );
};

Cards.propTypes = {
  books: PropTypes.arrayOf(
    PropTypes.shape({
      book_title: PropTypes.string.isRequired,
      book_author: PropTypes.arrayOf(PropTypes.string).isRequired,
      book_pages: PropTypes.number.isRequired,
      book_publication_city: PropTypes.string.isRequired,
      book_publication_country: PropTypes.string.isRequired,
      book_publication_year: PropTypes.number.isRequired,
      id: PropTypes.number.isRequired,
    })
  ).isRequired,
};

export default Cards;
