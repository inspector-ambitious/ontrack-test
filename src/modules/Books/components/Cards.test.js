import React from "react";
import { shallow } from "enzyme";
import Cards from "./Cards";

describe("Books/components/Cards", () => {
  const books = [
    {
      book_title: "Harry Potter and the Cursed Child",
      book_author: ["J. K. Rowling"],
      book_pages: 355,
      book_publication_city: "London",
      book_publication_country: "United Kingdom",
      book_publication_year: 2016,
      id: 1,
    },
  ];

  it("should render without crashing", () => {
    const wrapper = shallow(<Cards books={books} />);
    expect(wrapper).toMatchSnapshot();
  });
});
