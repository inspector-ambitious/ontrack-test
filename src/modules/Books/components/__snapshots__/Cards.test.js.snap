// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Books/components/Cards should render without crashing 1`] = `
<Row
  className="mt-4"
  noGutters={false}
>
  <Col
    className="d-flex align-items-stretch mb-4"
    key="1"
    lg={3}
    md={6}
    xs={12}
  >
    <Card
      body={false}
      className="flex-fill"
    >
      <CardBody>
        <CardTitle>
          Harry Potter and the Cursed Child
        </CardTitle>
      </CardBody>
      <ListGroup
        className="list-group-flush"
      >
        <ListGroupItem
          active={false}
          disabled={false}
        >
          Author: 
          J. K. Rowling
        </ListGroupItem>
        <ListGroupItem
          active={false}
          disabled={false}
        >
          Pages: 
          355
        </ListGroupItem>
        <ListGroupItem
          active={false}
          disabled={false}
        >
          Publication City: 
          London
        </ListGroupItem>
        <ListGroupItem
          active={false}
          disabled={false}
        >
          Publication Country: 
          United Kingdom
        </ListGroupItem>
        <ListGroupItem
          active={false}
          disabled={false}
        >
          Publication Year: 
          2016
        </ListGroupItem>
      </ListGroup>
    </Card>
  </Col>
</Row>
`;
